### ISSUE TITLE
Swift 6 warning: Main actor-isolated method cannot satisfy nonisolated protocol requirement

### ISSUE THREAD

**Issue by CatMe0w** [stadiamaps/ferrostar#164]
https://github.com/stadiamaps/ferrostar/blob/b8d925a991e04d529f275a36e1162441da07ef4a/apple/Sources/FerrostarCore/FerrostarCore.swift#L349

Xcode 16 gives this warning for this line in FerrostarCore package:

`Main actor-isolated instance method 'locationManager(_:didUpdateLocations:)' cannot be used to satisfy nonisolated protocol requirement; this is an error in the Swift 6 language mode`

**Comment by ianthetechie** [comment]
That's correct. It's a known limitation. As far as we are aware, the current code is safe, but due to a variety of reasons, it's not possible for us to satisfy the Swift 6 strict concurrency checker ;) Long-term we want to transition this to an actor type, but it's not high enough priority quite yet.

**Comment by ianthetechie** [comment]
We should probably coordinate on a timing of a full Swift 6 transition at some point. I don't think it's an urgent item as we can get pretty far with Swift 5.10, but we'll eventually need to switch.

Any opinions on this @Archdoog @hactar?

**Comment by ianthetechie** [comment]
(Also of note: I haven't looked into how far along the path UniFFI is, so there will still probably be a 5.10 module for a while)

**Comment by ianthetechie** [comment]
One more thing to note: the types in UniFFI are in fact marked `Sendable`, but only when using the Swift 6 compiler. So for the moment, the demo app marks these imports as `@preconcurrency`. We can revisit this later.

Import hacks added in #632.
